connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java

```diff
@@ -1154,7 +1152,17 @@ public void shouldParseFractionalPartsAsIntegerWhenNoFractionalPart() {
         assertEquals(new SchemaAndValue(Schema.INT32_SCHEMA, 66000), Values.parseString("66000.0"));
         assertEquals(new SchemaAndValue(Schema.FLOAT32_SCHEMA, 66000.0008f), Values.parseString("66000.0008"));
     }
+    @Test
+    public void avoidCpuAndMemoryIssuesConvertingExtremeBigDecimals() {
+    	long s = System.currentTimeMillis();
+        String veryBig = "1e+100000000"; // new BigDecimal().setScale(0, RoundingMode.FLOOR) takes around two minutes and uses 3GB;
+        System.out.println("time taken" + System.currentTimeMillis() - s)
+        assertEquals(new SchemaAndValue(Schema.BYTES_SCHEMA, veryBig), Values.parseString(veryBig));
```

Comment:
This shouldn't be a plain BYTES schema, it should be a connect `Decimal` type. And veryBig needs to be parsed into a BigDecimal.